h/groups/views.py

```diff
@@ -101,7 +105,28 @@ def _read_group(request, group):
 
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
-    template_data = {'group': group, 'group_url': url}
+
+    result = search.search(request, private=False, params={
+        "group": group.pubid, "limit": 1000})
+    annotations = result['rows']
+
+    # Group the annotations by URI.
+    # Create a dict mapping the (normalized) URIs of the annotated documents
+    # to the most recent annotation of each document.
+    annotations_by_uri = collections.OrderedDict()
```

Comment:
Matter of opinion I guess, I find my way easier to read. Mine also breaks as soon as it has 25 unique normalized URIs.